sympy/matrices/expressions/applyfunc.py

```diff
@@ -68,8 +68,16 @@ def expr(self):
     def shape(self):
         return self.expr.shape
 
-    def func(self, expr):
-        return ElementwiseApplyFunction(self.function, expr)
+    @property
+    def func(self):
+        # This strange construction is required by the assumptions:
+        # (.func needs to be a class)
+
+        class ElementwiseApplyFunction2(ElementwiseApplyFunction):
```

Comment:
@asmeurer do we have a better way to handle this `.func`? `.func` needs to be a class otherwise `sympy.assumptions` complains.

This class is intended to encapsulate elementwise function application to matrices. The function is saved as `self.function`, I wanted `.func` to accept the same parameters as `self.function.func`.